tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+import json
+import os
+import time
+import boto3
+import pandas
+import pytest
+from awsglue.context import GlueContext
+from pyspark.sql.session import SparkSession
+from src.glue.jobs import json_to_parquet
+from unittest.mock import patch
+# requires pytest-datadir to be installed
+
+
+@patch(
+        "src.glue.jobs.json_to_parquet.INDEX_FIELD_MAP",
+        {"TestFlatDataType": ["GlobalKey"], "TestNestedDataType": ["GlobalKey"]}
+)
+class TestJsonS3ToParquet:
+
+    @pytest.fixture(scope="class")
+    def glue_database_name(self, namespace):
+        return f"{namespace}-pytest-database"
+
+    @pytest.fixture(scope="class")
+    def glue_nested_table_name(self):
+        return "dataset_TestNestedDataType"
+
+    @pytest.fixture(scope="class")
+    def glue_flat_table_name(self):
+        return "dataset_TestFlatDataType"
+
+    @pytest.fixture(scope="class")
+    def workspace_key_prefix(self, namespace, artifact_bucket):
+        workspace_key_prefix = os.path.join(
+            namespace,
+            "tests/test_json_to_parquet"
+        )
+        yield workspace_key_prefix
+        s3_client = boto3.client("s3")
+        response = s3_client.list_objects_v2(
+                Bucket=artifact_bucket,
+                Prefix=workspace_key_prefix
+        )
+        if "Contents" in response:
+            objects_to_delete = []
+            for obj in response["Contents"]:
+                objects_to_delete.append({"Key": obj["Key"]})
+            if objects_to_delete:
+                s3_client.delete_objects(
+                    Bucket=artifact_bucket,
+                    Delete={"Objects": objects_to_delete}
+                )
```

Comment:
When you yield here, does the rest of this code execute?

Reply:
This is pytest's way of doing setup/teardown for fixtures. In this fixture I define the workspace (S3) location and scope it to the class. When the class (e.g., the entire test suite for JSON to Parquet) completes, I clean up the workspace using the post-yield code.

Reply:
TIL: This, very neat.